cmd/kaspawallet/sweep.go

```diff
@@ -0,0 +1,267 @@
+	}
+
+	daemonClient, tearDown, err := client.Connect(conf.DaemonAddress)
+	if err != nil {
+		return err
+	}
+	defer tearDown()
+
+	ctx, cancel := context.WithTimeout(context.Background(), daemonTimeout)
+	defer cancel()
+
+	getExternalSpendableUTXOsResponse, err := daemonClient.GetExternalSpendableUTXOs(ctx, &pb.GetExternalSpendableUTXOsRequest{
+		Address: address.String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	UTXOs, err := libkaspawallet.KaspawalletdUTXOsTolibkaspawalletUTXOs(getExternalSpendableUTXOsResponse.Entries)
+	if err != nil {
+		return err
+	}
+
+	paymentAmount := uint64(0)
+
+	if len(UTXOs) == 0 {
+		return errors.Errorf("Could not find any spendable UTXOs in %s", addressPubKey)
+	}
+
+	for _, UTXO := range UTXOs {
+		paymentAmount = paymentAmount + UTXO.UTXOEntry.Amount()
+	}
+
+	newAddressResponse, err := daemonClient.NewAddress(ctx, &pb.NewAddressRequest{})
+	if err != nil {
+		return err
+	}
+
+	toAddress, err := util.DecodeAddress(newAddressResponse.Address, conf.ActiveNetParams.Prefix)
+	if err != nil {
+		return err
+	}
+
+	splitTransactions, err := createSplitTransactionsWithSchnorrPrivteKey(conf.NetParams(), UTXOs, toAddress, feePerInput)
+	if err != nil {
+		return err
+	}
+
+	serializedSplitTransactions, err := signWithSchnorrPrivteKey(conf.NetParams(), privateKeyBytes, splitTransactions)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("\nSweeping...")
+	fmt.Println("\tFrom:\t", addressPubKey)
+	fmt.Println("\tTo:\t", toAddress)
+
+	response, err := daemonClient.Broadcast(ctx, &pb.BroadcastRequest{
+		IsDomain:     true,
+		Transactions: serializedSplitTransactions,
+	})
+	if err != nil {
+		return err
+	}
+
+	totalExtracted := uint64(0)
+
+	fmt.Println("\nTransaction ID(s):")
+	for i, txID := range response.TxIDs {
+		fmt.Printf("\t%s\n", txID)
+		fmt.Println("\tSweeped:\t", utils.FormatKas(splitTransactions[i].Outputs[0].Value), " KAS")
+		totalExtracted = totalExtracted + splitTransactions[i].Outputs[0].Value
+	}
+
+	fmt.Println("\nTotal Funds Sweeped (including transaction fees):")
+	fmt.Println("\t", utils.FormatKas(totalExtracted), " KAS")
+
+	return nil
+}
+
+func newDummyTransaction() *externalapi.DomainTransaction {
+	return &externalapi.DomainTransaction{
+		Version:      constants.MaxTransactionVersion,
+		Inputs:       make([]*externalapi.DomainTransactionInput, 0), //we create empty inputs
+		LockTime:     0,
+		Outputs:      make([]*externalapi.DomainTransactionOutput, 1), // we should always have 1 output to the toAdress
+		SubnetworkID: subnetworks.SubnetworkIDNative,
+		Gas:          0,
+		Payload:      nil,
+	}
+}
+
+func createSplitTransactionsWithSchnorrPrivteKey(
+	params *dagconfig.Params,
+	selectedUTXOs []*libkaspawallet.UTXO,
+	toAddress util.Address,
+	feePerInput int) ([]*externalapi.DomainTransaction, error) {
+
+	var splitTransactions []*externalapi.DomainTransaction
+
+	// I need to add extra mass to transaction, txmasscalculater seems to undercalculate, by around this amount.
```

Comment:
It doesn't undercalculate. You just don't take signatures into account. Please rephrase the comment to reflect it. But anyway, please try to use `we` instead of `I`.